### PR TITLE
adds caching to stats endpoints

### PR DESCRIPTION
**Pull request by kenodressel**

### CODE REVIEW

**Review by thepiwo — COMMENTED** [review]
I don't think extrashort is needed if we invalidate well

File: models/retip.js [hunk 1]
```diff
@@ -16,5 +17,7 @@ module.exports = (sequelize, DataTypes) => {
     timestamps: true,
   });
   Retip.belongsTo(Tip(sequelize, DataTypes), { foreignKey: 'tipId' });
+  Retip.addHook('beforeCreate', async () => { await cache.del(['fetchStats']); });
+  Retip.addHook('beforeUpdate', async () => { await cache.del(['fetchStats']); });
```
Comment (thepiwo) [inline models/retip.js:21]: not after?

File: models/tip.js [hunk 2]
```diff
@@ -1,27 +1,33 @@
...
+      allowNull: false,
+    },
+    unclaimed: {
+      type: DataTypes.BOOLEAN,
+      defaultValue: true,
+    },
+  }, {
+    timestamps: true,
+  });
+  Tip.addHook('beforeCreate', async () => { await cache.del(['fetchStats']); });
```
Comment (thepiwo) [inline models/tip.js:30]: same here, not delete after db stuff happened?

Comment (thepiwo) [reply models/tip.js:30]: also you can probably skip the async ... await?

File: models/retip.js (hunk 1 above)
Comment (kenodressel) [reply models/retip.js:21]: does generally not make much of a difference since we are using the tips cache anyways but for readability it makes more sense

**Review by thepiwo — APPROVED** [review]

**Pull request merged**